lib/iris/tests/test_analysis.py

```diff
@@ -351,102 +351,170 @@ def test_multi_coord_mdtol(self):
 
 
 class TestAggregators(tests.IrisTest):
-    def test_percentile_1d(self):
+
+    def _check_coord_properties(self, cube, collapse_coord, unit, points):
+        if not isinstance(collapse_coord, list):
+            collapse_coord = [collapse_coord]
+        name = 'percentile_over_' + '_'.join(collapse_coord)
+        self.assertTrue(cube.coord(name))
+        self.assertEqual(cube.coord(name).units, unit)
+        self.assertEqual(cube.coord(name).points, points)
+        
+    def _check_collapsed_percentile(self, cube, percents, collapse_coord,
+                                    expected_result, coord_check=False,
+                                    **kwargs):
+
+        expected_result = np.array(expected_result, dtype=np.float32)
+        result = cube.collapsed(collapse_coord, iris.analysis.PERCENTILE,
+                                percent=percents, **kwargs)
+        np.testing.assert_array_almost_equal(result.data, expected_result)
+        if coord_check:
+            self._check_coord_properties(result, collapse_coord, 1, [percents])
+
+    def _check_percentile(self, data, axis, percents, expected_result,
+                          coord_check=False, **kwargs):
+
+        result = iris.analysis._percentile(data, axis, percents, **kwargs)
+
+        np.testing.assert_array_almost_equal(result, expected_result)
+
+    def test_percentile_invalid_percentile_method(self):
```

Comment:
Implementing my suggested change above would remove the need for this test method too: there can only be two options and one of them will always be selected.

Reply:
:white_check_mark: